syndicate/core/build/warmup_processor.py

```diff
@@ -0,0 +1,296 @@
+                                                        httpMethod=method)
+            if 'uri' in integration:
+                lambda_uri = integration['uri']
+                if lambda_uri not in affected_lambda:
+                    affected_lambda.append(lambda_uri)
+                    api_gw_path = resource_items_path[resource_id]
+                    if not api_gw_path in allowed_path_method:
+                        allowed_path_method.update({api_gw_path: [method]})
+                    else:
+                        allowed_path_method[api_gw_path].append(method)
+    return allowed_path_method
+
+
+def process_inputted_api_gw_id(api_id, stage_name):
+    api_gw_client = _get_api_gw_client()
+    all_apis = api_gw_client.get_rest_apis().get('items', {})
+
+    allowed_id = []
+    for api in all_apis:
+        allowed_id.append(api['id'])
+
+    schemas_list = []
+    paths_to_be_triggered = {}
+    for rest_api_id in api_id:
+        if rest_api_id not in allowed_id:
+            click.echo(f'Provided {rest_api_id} API ID does not exists')
+            continue
+
+        allowed_api_id = get_api_stages(rest_api_id, stage_name)
+        for id, api_gw_meta in allowed_api_id.items():
+            for meta in api_gw_meta:
+                paths_to_be_triggered.update({id: get_api_gw_integration(id)})
+                schema = transform_to_schema(meta)
+                schemas_list.append(schema)
+    return schemas_list, paths_to_be_triggered
+
+
+def load_schema(api_gw_resources_meta):
+    schemas = []
+    paths_to_be_triggered = {}
+    for resource_arn, meta in api_gw_resources_meta.items():
+        rest_api_id = resource_arn.split('/')[-1]
+        stage_name = meta.get('resource_meta', {}).get('deploy_stage')
+
+        exported_schema = get_api_gw_export(rest_api_id, stage_name)
+        paths_to_be_triggered.update(
+            {rest_api_id: get_api_gw_integration(rest_api_id)}
+        )
+        schema = transform_to_schema(exported_schema)
+        schemas.append(schema)
+    return schemas, paths_to_be_triggered
+
+
+@exit_on_exception
+def warmup_resources(bundle_name, deploy_name):
+    output = load_deploy_output(bundle_name, deploy_name)
+
+    filters = [
+        lambda v: v['resource_meta'].get('resource_type') == 'api_gateway'
+    ]
+
+    for function in filters:
+        output = _filter_the_dict(dictionary=output, callback=function)
+
+    if not output:
+        _LOG.warning('No resources to warmup, exiting')
+        return
+
+    schemas = load_schema(api_gw_resources_meta=output)
+    return schemas
+
+
+def find_api_url(schema_doc):
+    server = schema_doc['servers'][0]
+    api_base_path = server['variables']['basePath']['default'][1:]
+    url = server['url'].format(basePath=api_base_path)
+    return url
+
+
+def find_api_methods(schema_doc):
+    url = find_api_url(schema_doc)
+    paths = schema_doc['paths']
+    url_method_dict = {}
+    url_method_dict.update({url: paths})
```

Comment:
why you instantiate url_method_dict as an empty dict and then update it, there is no sense of it